base/abstractset.jl

```diff
@@ -476,6 +476,62 @@ function isdisjoint(a, b)
     _isdisjoint(a, b)
 end
 
+function isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T<:Union{Real,Char}
```

Comment:
My thought was to limit it to types that are tested, to avoid the possibility of incorrect results on an untested type with subtly different behavior. Was that overly cautious? (Or am I betraying a lack of confidence in the general correctness of my code? :stuck_out_tongue_closed_eyes: )